src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -265,8 +268,14 @@ static void Initialize ()
 						typeof (Action<AppDomain, UnhandledExceptionEventArgs>), ad_due);
 				}
 			}
+#endif
 		}
 
+#if NETCOREAPP
+		[DllImport (AndroidRuntime.InternalDllName, CallingConvention = CallingConvention.Cdecl)]
+		extern static void monodroid_unhandled_exception (Exception javaException);
```

Comment:
If passing a managed type like this, doesn't this need to be an internal call, not a DllImport?

Reply:
I think it's treated as an `IntPtr` in this case, it appears to work with the test app

Reply:
@vargaz thinks it should be an icall: https://discord.com/channels/732297728826277939/732297965019988138/867104696488427580

It might "work", but I don't think it's *guaranteed* to work.

Reply:
Done